File: utils/launcher_settings.go
/**
 * This script allows you to update the gameDataPath in the
 * launcher-settings.json file in case you moved your game folder.
 */

package utils

import (
	"encoding/json"
	"fmt"
	"os"
)

func UpdateGameDataPath(filePath, newGameDataPath string) error {
	data, err := decodeJsonFile(filePath)
	if err != nil {
		return err
	}

	data["gameDataPath"] = newGameDataPath

	return encodeJsonFile(filePath, data)
}

func decodeJsonFile(filePath string) (map[string]interface{}, error) {
	file, err := os.Open(filePath)
	if err != nil {
		return nil, err
	}

	defer func() {
		closeErr := file.Close()
		if closeErr != nil {
			fmt.Println("Failed to close the file:", closeErr)
		}
	}()

	var data map[string]interface{}
	decoder := json.NewDecoder(file)
	if err := decoder.Decode(&data); err != nil {
		return nil, err
	}

	return data, nil
}

func encodeJsonFile(filePath string, data map[string]interface{}) error {
	file, err := os.Create(filePath)
	if err != nil {
		return err
	}

	defer func() {
		closeErr := file.Close()
		if closeErr != nil {
			fmt.Println("Failed to close the file:", closeErr)
		}
	}()

	encoder := json.NewEncoder(file)
	encoder.SetIndent("", "\t")

	return encoder.Encode(data)
}

File: src/scripts/launcher-settings.js
const fs = require('fs');
const path = require('path');

async function readFile(filePath, encoding) {
  return new Promise((resolve, reject) => {
    fs.readFile(filePath, encoding, (err, data) => {
      if (err) {
        reject(err);
      } else {
        resolve(data);
      }
    });
  });
}

async function writeFile(filePath, data) {
  try {
    await fs.promises.writeFile(path.normalize(filePath), data);
  } catch (error) {
    console.error(`Error writing to file: ${error}`);
  }
}

async function updateGameDataPath(filePath, newGameDataPath) {
  try {
    const data = await readFile(filePath, 'utf8');
    const jsonData = JSON.parse(data);

    // Update the gameDataPath property
    jsonData.gameDataPath = newGameDataPath;

    // Write the updated JSON file
    await writeFile(filePath, JSON.stringify(jsonData, null, 2));
    console.log('Game data path successfully updated in launcher-settings.json!');
  } catch (error) {
    console.error(`Error updating game data path: ${error}`);
  }
}

module.exports = {
  updateGameDataPath
};
